zips/zip-0032.rst

```diff
@@ -397,18 +402,24 @@ $\mathsf{MKGh}^\mathsf{Context}(\mathsf{IKM}) \rightarrow (\mathsf{sk}_m, \maths
 Hardened-only child key derivation
 ----------------------------------
 
-$\mathsf{CKDh}^\mathsf{Context}((\mathsf{sk}_{par}, \mathsf{c}_{par}), i)$ $\rightarrow (\mathsf{sk}_i, \mathsf{c}_i)$ :
+As well as the integer child index $i$, the child key derivation function defined here supports a
+byte sequence $\mathsf{tag}$ and a $\mathsf{full\_width\_leaf}$ flag ($\!0$ representing $\mathsf{false}$,
+$1$ representing $\mathsf{true}$). Each triple $(i, \mathsf{tag}, \mathsf{full\_width\_leaf})$ produces
+a different output.
 
-- Check whether $i \geq 2^{31}$ (whether the child is a hardened key).
+The $\mathsf{tag}$ and $\mathsf{full\_width\_leaf}$ inputs are reserved for use by `Registered child key derivation`_;
+for other uses $\mathsf{tag}$ MUST be $[\,]$ and $\mathsf{full\_width\_leaf}$ MUST be $\mathsf{false}$ ($\!0$).
 
-  - If so (hardened child): let
-    $I = \mathsf{PRF^{expand}}(\mathsf{c}_{par}, [\mathsf{Context.CKDDomain}]\,||\,\mathsf{sk}_{par}\,||\,\mathsf{I2LEOSP}_{32}(i))$.
-  - If not (normal child): return failure.
+$\mathsf{CKDh}^\mathsf{Context}((\mathsf{sk}_{par}, \mathsf{c}_{par}), i, \mathsf{tag}, \mathsf{full\_width\_leaf})$ $\rightarrow (\mathsf{sk}_{child}, \mathsf{c}_{child})$ :
```

Comment:
Why is the argument order `tag, full_width_leaf` instead of the reverse? The `full_width_leaf` flag needs to be better documented in general; its meaning is not sufficiently clarified in the `Registered child key derivation` section; all that's described there is its use.

Reply:
`full_width_leaf` only occurs as an argument to `CKDh`; putting it last allows `i, tag` to be together in all cases, which is logically correct since they together specify the index. The usage of `full_width_leaf` is now documented.